src/reference_implementations/prompting_vector_llms/llm_prompting_examples/llm_prompt_ag_news.ipynb

```diff
@@ -0,0 +1,1107 @@
+    "    generation = model.generate(prompt_batch, short_generation_config)\n",
+    "    # We'll use tokens this time and consider just the first token\n",
+    "    first_predicted_tokens = [tokens[0].strip().lower() for tokens in generation.generation[\"tokens\"]]\n",
+    "    # If a token doesn't correspond to one of our labels, we'll randomly select one\n",
+    "    for potential_prediction in first_predicted_tokens:\n",
+    "        if potential_prediction in lowercase_labels:\n",
+    "            predicted_labels.append(potential_prediction)\n",
+    "        else:\n",
+    "            unmatched_predictions.append(potential_prediction)\n",
+    "            predicted_labels.append(random.choice(lowercase_labels))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Example Prompt\n",
+      "Title: Telecom lifts first quarter net profit 19pc\n",
+      "Description: Telecom Corp today reported its September first quarter net profit rose 19 per cent to $193 million. The profit bettered analysts #39; average forecasts of $185m.\n",
+      "From World, Sports, Business, Technology, what category does this article belong to? \n",
+      "---------------------------------------------\n",
+      "Failed to match 100 responses to our label space\n",
+      "Some examples of responses: ['1', '0', '0', '1', '>>', '�', '5', '1', '¶', '0']\n",
+      "Prediction Accuracy: 0.26\n",
+      "Confusion Matrix with ordering ['world', 'sports', 'business', 'technology']\n",
+      "[[ 7 10  7  4]\n",
+      " [ 7  6  7  1]\n",
+      " [ 5  4  7  7]\n",
+      " [11  7  4  6]]\n",
+      "========================================================\n",
+      "Label: world, F1: 0.2413793103448276, Precision: 0.23333333333333334, Recall: 0.25\n",
+      "Label: sports, F1: 0.25, Precision: 0.2222222222222222, Recall: 0.2857142857142857\n",
+      "Label: business, F1: 0.2916666666666667, Precision: 0.28, Recall: 0.30434782608695654\n",
+      "Label: technology, F1: 0.2608695652173913, Precision: 0.3333333333333333, Recall: 0.21428571428571427\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"Example Prompt\\n{prompts[0]}\")\n",
+    "print(\"---------------------------------------------\")\n",
+    "print(f\"Failed to match {len(unmatched_predictions)} responses to our label space\")\n",
+    "print(f\"Some examples of responses: {sample(unmatched_predictions, 10)}\")\n",
+    "report_metrics(predicted_labels, ag_news_labels, labels_order=[\"world\", \"sports\", \"business\", \"technology\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see that the model actually never answers with a response in our expected label space. As a result, we end up randomly guessing for all of our predictions, leading to random accuracy."
```

Comment:
I feel like "random accuracy" might be a bit misleading 😂

Reply:
That's fair 😂